ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/resources/impl/ResourceManager.java

```diff
@@ -544,7 +544,9 @@ private NewProjectConfigDto asDto(MutableProjectConfig config) {
         checkArgument(!source.getLocation().isRoot(), "Workspace root is not allowed to be copied");
```

Comment:
In this class we have many validations like this:
`checkArgument(expression, errorMessage)`
So, I think we should change many of them but not in this issue.